xaynetwork/xayn_discovery_app
# TB-4068 Share document using deep link

### What 🕵️ 🔍

- What does the pull request solve?
Introduces the possibility to share a document by using deep link and therefore to open the shared document within the app after clicking on the deep link.
This PR covers the scenario where the user, clicking the deep link, has the app installed
- What exactly was added or modified?

----------

### How to test (please adjust template) 🥼 🔬

Please test for Android and iOS.
To be tested with the internal build and not in debug mode

First scenario:

- Share a document by using the share option. You will have a link that you can click.
- Tap on the link and:
   - if the app is in background, check that the app is opened and a new page is pushed, on top of the previous one, containing the shared article in reader mode
   - if the app is closed, checked that the app is launched and that the page containing the shared article in reader mode. In this scenario, check that when going back from the current article view you go back to the feed

Second scenario:

- Uninstall the app
- Tap on the link and check that the browser is opened and you are redirected to the web article viewer. (Please notice that at the moment only firefox seems to allow the redirect correctly. Other browsers open the store. EDIT: this should now work for every browser

Third scenario:

- Open the link in any browsers on your laptop and check that you are redirected to the web article viewer

----------

### Screenshots 📸 📱

| Before | After  |
| ------ | ------ |
| <img width="280" src=""> | <img width="280" src=""> |

----------

### References 📝 🔗

- [JIRA](https://xainag.atlassian.net/browse/TB-4068)

----------
